src/camera.c

```diff
@@ -126,7 +126,7 @@ query_permission_sync (Request *request)
 
       if (!xdp_dbus_impl_access_call_access_dialog_sync (access_impl,
                                                          request->id,
-                                                         app_id,
+                                                         info ? app_id : "",
```

Comment:
I wonder if it should be considered whether there is a race here:

1. Install flatpak
2. Run flatpak before .desktop file has been discovered by xdg-desktop-portal
3. Circumvent access

Reply:
Ok. Checking whether `app_id` is valid should be also enough and there shouldn't be a race you described.